### PR TITLE
Revert allocation transaction committed

### PR DESCRIPTION
**Pull request by jbaublitz**
Related to #3274

### CODE REVIEW

**Review by mulkieran — CHANGES_REQUESTED** [review]
This all looks good. One minor docs request.

File: src/engine/strat_engine/backstore/data_tier.rs [hunk 1]
```diff
@@ -91,18 +90,20 @@ impl DataTier {
     /// Allocate a region for all sector size requests from unallocated segments in
     /// block devices belonging to the data tier. Return Some(_) if requested
```
[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/engine/strat_engine/backstore/data_tier.rs]: Resolved.

**Comment by jbaublitz** [comment]
I also made one PR revision that appeared to be required to avoid a panic in cases where we incorrectly calculate the amount of space to allocate from the backstore to the thinpool. I bumped into this due to a bug in #3274 that I intend to fix but it seems better to handle that case with an error in case we introduce another bug like that in the future.

**Comment by jbaublitz** [comment]
Putting back in progress to look into one additional piece brought up by #3274.

**Comment by jbaublitz** [comment]
I think this is ready for final review.

**Comment by mulkieran** [comment]
@jbaublitz Plz rebase when you get the chance.

**Review by mulkieran — APPROVED** [review]

**Comment by mulkieran** [comment]
@jbaublitz Plz rebase when you get the chance and we can merge this.

**Pull request merged**